### DISCUSSION TITLE
dispatchSync without get the models from db

### DISCUSSION THREAD

**Discussion by sbc640964** [laravel/framework#43551] [Ideas]
Set the synchronous queues that will not restore the database models

**Comment by jessealkema** [comment]
@sbc640964 By removing the `SerializesModels` trait the models will not be reloaded when the job is executed.

**Reply by sbc640964** [reply]
But I don't want to cancel them.
I just want to cancel them in the synchronous method.

**Reply by jessealkema** [reply]
You can always write your own `SerializesModels` trait that checks the current queue connection for the job. If the connection is another type then sync the normal way is executed. When in sync the serialization of models is skipped. Of course your post is to enable this behavior by default in normal, just trying to help out.

Another tactic to come to the same end result is writing a service that handles the logic. That service is called from the job and the place you want to execute the code immediately with the manipulated data customized `with` methods to load specific relational data.

At least that's how I always work around these types of scenarios.
